### ISSUE TITLE
organization of test files

### ISSUE THREAD

**Issue by kain88-de** [MDAnalysis/mdanalysis#466]
Would it be ok to split up the test-files? Currently some of them are really huge (several thousand lines) and they just keep growing with random additions. I find it really hard to get an overview in these large files. I also have a bad overview which files from the package are tested well. As an example take `test_coordinates.py`. it contains  a lot of doublicated code. It's hard to tell just by looking at the file if a specific reader is tested well or not.

I personally would like a structure like this

```
MDAnalysisTests
   | -- coordinates
             base.py
             test_pdb.py
             test_dcd.py
             test_xdr.py
             ...
```

In such a file structure I find it easier to find tests for specific modules. Individual files aren't monolithic and just by the line-count one can get an impression how well tested individual modules are. 

I'm asking because I will rewrite some of the test_coordinates any way now. I'm doing this because @richardjgowers pointed out in #441 that my xtc implementation is set to the wrong frame after initialization and no test catches that. To me this seems like a test that every reader should have and pass. So I want to write some generic reader/writer test-classes that will make it easier to add such tests fast for all readers. At the some time I would also address #440 for the coordinates tests.
## Files to split up
- [x] test_coordinates.py [#469]
- [ ] test_atomgroup.py
- [x] test_analysis.py [#492]

**Comment by richardjgowers** [comment]
Yeah I think Readers (and Writers and Parsers) should have 90% of their tests come from some sort of base class.  And yeah, some files like test_atomgroup and test_coordinates are just a jungle inside.  I think the file structure you propose is good.

If you're going to look at Reader tests, check also `test_reader_api`.  Hopefully that can act as a start point for the required tests for each Reader.

**Comment by richardjgowers** [comment]
We could also probably make better use of the tagging system for tests.  If we split up test_coordinates, it would be useful to be able to test all Readers with a single command... But maybe this is equally possible through just running all the tests in the coordinates directory.

**Comment by orbeckst** [comment]
I am very much in favor of splitting the test files!

**Comment by orbeckst** [comment]
#492 split test_analysis.py

**Comment by orbeckst** [comment]
@richardjgowers & @dotsdl : Will #363 also refactor `test_atomgroup.py` and lead to closure on this issue?

**Comment by dotsdl** [comment]
It should. Although @richardjgowers can provide more details, most of the content of `AtomGroup` is determined from the topology, so it makes more sense that there are `groups` tests for conversions between `AtomGroups`, `ResidueGroups`, and `SegmentGroups`, and anything that's found in `GroupBase` itself. I think there should then be a `test_topologyattrs` with tests for each one of these, most of which can be done with some nice subclassing.

In short, the structure of the tests should match pretty closely to the structure of `core`, which means that yes, there will be a reorganization.

**Comment by richardjgowers** [comment]
Yeah I think test atomgroup will get rewritten completely. Probably a core
subdirectory happening too. Woo tests

**Comment by kain88-de** [comment]
> Probably a core subdirectory happening too

:+1: that sounds really nice. it will also make it much easier to find tests
